src/libs/ReportUtils.ts

```diff
@@ -3399,11 +3426,63 @@ function canAccessReport(report: OnyxEntry<Report>, policies: OnyxCollection<Pol
 function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): boolean {
     const currentReport = getReport(currentReportId);
     const parentReport = getParentReport(isNotEmptyObject(currentReport) ? currentReport : null);
-    const reportActions = ReportActionsUtils.getAllReportActions(report?.reportID ?? '');
-    const isChildReportHasComment = Object.values(reportActions ?? {})?.some((reportAction) => (reportAction?.childVisibleActionCount ?? 0) > 0);
+    const allReportActions = ReportActionsUtils.getAllReportActions(report?.reportID ?? '');
+    const isChildReportHasComment = Object.values(allReportActions ?? {})?.some((reportAction) => (reportAction?.childVisibleActionCount ?? 0) > 0);
     return parentReport?.reportID !== report?.reportID && !isChildReportHasComment;
 }
 
+/**
+ *  Checks if there are any violations belonging to the transaction in the transactionsViolations Onyx object
+ *  then checks that the violation is of the proper type
+ */
+function transactionHasViolation(transactionID: string): boolean {
+    const violations = transactionViolations ? transactionViolations[transactionID]?.value : [];
+    if (!violations) {
+        return false;
+    }
+    return violations.some((violation: TransactionViolation) => violation.type === 'violation');
+}
+
+/**
+ *  Checks to see if a report's parentAction is a money request that contains a violation
+ *  This only applies to report submitter and for reports in the open and processing states
+ */
+
+function transactionThreadHasViolations(report: Report, betas: Beta[]): boolean {
```

Comment:
I think we can use `usePermissions` hook in view components and no need to pass here